### ISSUE TITLE
old closure super ctor pattern throws off incremental clutz

### ISSUE THREAD

**Issue by rkirov** [angular/clutz#639]
The following

```javascript
goog.provide('a.b.C');

/**
 * @constructor
 * @extends {Foo}
 */
a.b.C = function() {
    a.b.C.base(this, 'constructor');

    /**
     * @type {string}
     */
    this.missing;

    /**
     * @type {string}
     */
    this.emitted = '';
};
```

Results in `missing` not being emitted.

**Comment by evmar** [comment]
Looks like this also needs:

```
// at the top:
goog.require('goog.base');

// at the bottom:
goog.inherits(a.b.C, Foo);
```

**Comment by LucasSloan** [comment]
I spent a day trying to get closure to output the right thing for us and failed.  I talked it over with Evan, and he suggested that the example is missing a call to `goog.extends` or something like that.  I still need to follow up and see if that's correct.

**Comment by rkirov** [comment]
Modified the code snippet to have the goog.inherits calls as original call. Issue still present.

**Comment by rkirov** [comment]
Full isolated repro - https://github.com/rkirov/clutz/commit/466c80fa8bae746162d481c87cd70e11341bfdd2

I don't see anything suspicious in clutz code here, when the debugger is hooked up. Generally closure does not report those properties. Curiously enough if the base class is visible or removed the field is seen again.

**Comment by rkirov** [comment]
To make matters worse, in the particular case I am working against

```
 /**
   * @type {string|undefined}
   */
  this.missing = undefined;
```
results in ```missing: undefined```, because the actual string assigment is outside the visibility of the incremental clutz run (outside the class definition).

**Comment by evmar** [comment]
What's the suggested workaround, initializing the fields?

**Comment by rkirov** [comment]
Yes, but they you might run into a type narrowing issue above. Where we emit 

```
/**
   * @type {string|undefined}
   */
  this.missing = undefined;
```

as ``` missing: undefined```, unless clutz sees a string assignment somewhere. The workaround for that is:
```
C.prototype.clutzWorkaroundFn = {
  this.missing = '';
}
```
which would be optimized away.

**Comment by rkirov** [comment]
This will be fixed in closure, keeping the bug open to track adding tests and syncing new compiler version.
